lib/Badge.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Material badge for showing notifications, maybe more functionality in the future
+ * @flow
+ */
+
+import React, { Component } from 'react'
+import {
+  View,
+  Text,
+  StyleSheet
+} from 'react-native';
+
+type BProps = {
+  text: string,
+  size: number,
+  children: Array<ReactElement<*>>,
+  style: any,
+  isVisible: boolean,
+}
+
+const defaultProps = {
+  size: 20,
+  isVisible: true,
+  style: {},
+}
+
+export default class RippleBackgroundTransition extends Component {
+  static defaultProps: typeof defaultProps
+  props: BProps
+  static defaultProps = defaultProps
+
+  createStyles() {
+    const { size, style } = this.props;
+    const container = Object.assign(
+      {},
+      {
+        position: "relative",
+        width: size,
+        height: size,
+        borderRadius: size / 2,
+        alignItems: "center",
+        justifyContent: "center",
+        backgroundColor: "#F50057",
+        zIndex: 9999,
+        bottom: 5,
+        right: 10,
+      },
+      style.container
+    );
+
+    const text = Object.assign(
+      {},
+      {
+        color: "#fff",
+        fontWeight: "500",
+        fontSize: 12,
+      },
+      style.text
+    )
+
+    return StyleSheet.create({
+      container,
+      text,
+    })
+  }
+
+  render() {
+    const { children, text, isVisible } = this.props;
+    const styles = this.createStyles();
+    if (!isVisible) {
```

Comment:
_For android compatibility:_ Remove the isVisible check, lines 70-72. The check is already done in Tab.js

Reply:
So the check `badgeText === undefined && !isBadgeVisible` is to prevent the badge from rendering when there is no text. The check in here is for the user to be able to hide a badge even when there is text.

Reply:
Ok, I see.